packages/salesforcedx-vscode-core/src/commands/forceSourceDeploySourcePath.ts

```diff
@@ -33,6 +44,62 @@ export class ForceSourceDeploySourcePathExecutor extends BaseDeployExecutor {
     return commandBuilder.build();
   }
 
+  public async execute(response: ContinueResponse<string>): Promise<void> {
+    const betaPerfEnabled = sfdxCoreSettings.getBetaPerfEnhancements();
+    // this supported types logic is temporary until we have a way of generating the metadata type from the path
+    // once we have the metadata type we can check to see if it is a toolingsupportedtype from that util
+    const supportedType =
+      path.extname(response.data) === APEX_CLASS_EXTENSION ||
+      response.data.includes(`${APEX_CLASS_EXTENSION}-meta.xml`);
+    if (betaPerfEnabled && supportedType) {
+      const startTime = process.hrtime();
+      const cancellationTokenSource = new vscode.CancellationTokenSource();
+      const cancellationToken = cancellationTokenSource.token;
+
+      const executionWrapper = new CompositeCliCommandExecutor(
+        this.build(response.data)
```

Comment:
Using the default `build()` here give the impression in the output panel that we're still running the CLI command

Reply:
Yeah, but to avoid doing that we would need to update `CompositeCliCommandExecutor`'s constructor or use a new executor class. I think it's ok for now but we should address it in the next PRs.